ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -481,23 +486,17 @@ public void removeImportedTransaction(Xid xid) throws XAException {
         SubordinationManager.getTransactionImporter().removeImportedTransaction(xid);
     }
 
-    public Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, String parentNodeName, int recoveryFlags) throws XAException {
+    public Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, String parentNodeName, int recoveryFlags) throws XAException, UnsupportedOperationException {
```

Comment:
this change was asked by Tom, but I agree to revert it back (the same way in SPI) and leave it as it was